### PR TITLE
Add LeptonicHiggsBuilder and refactor

### PR DESCRIPTION
**Pull request by vvolkl**
Adds a class named LeptonicHiggsBuilder, needed for the `h4l` analysis, and tries to refactor the code of `LeptonicZedBuilder` and `ResonanceBuilder`. Note that I tried to keep the behavior of LeptonicZedBuilder unchanged - if it's ok to make some minor changes to the interface the code in heppy could be a bit cleaner.

### CODE REVIEW

**Comment by cbernet** [comment]
Hi @vvolkl! Can we find some time to discuss this PR today? please contact me on skype (firstname.lastname)

File: analyzers/ResonanceBuilder.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import pprint 
 import itertools
 
-mass = {23: 91, 25: 125}
+mass = {23: 91, 25: 125, 32:200}
```
Comment (clementhelsens) [inline analyzers/ResonanceBuilder.py]: For resonnances that don't have a fix mass, like 32 (Z') we should be able to overwrite the mass value in the analyser configuration

Comment (cbernet) [reply analyzers/ResonanceBuilder.py]: hello, no need to keep reviewing until  i discuss with Valentin, I think we need to go in a different direction. no big deal and I can write the few lines needed, so no worries :-)

Comment (vvolkl) [reply analyzers/ResonanceBuilder.py]: Yes @clementhelsens, there is now a parameter `nominal_mass`, this dict is just used if it is not present in order not to break any existing user code. But let's see what the results of the discussion with Colin are - @cbernet, I wrote you on skype :-)

**Comment by cbernet** [comment]
As discussed, I'm closing. We discussed to easy alternative solutions that are much more generic.

**Pull request closed without merging**